### PR TITLE
[pvr] fix EPG crash and recording errors produced by not present data…

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description
Found during my rework a crash produced by addon call of `PVR::CPVRClient::IsRecordable(std::shared_ptr<PVR::CPVREpgInfoTag const> const&, bool&) const`
where "[strGenreDescription](https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_epg_types.h#L99)" on EPG_TAG was not set and has produced a crash by try to read them.

Further I seen also on PVR_RECORDING some values not set from Kodi by call to addon with them.

This change here has fixed the faults by me.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/pvr/recordings/PVRRecording.h [hunk 1]
```diff
@@ -287,13 +287,39 @@ namespace PVR
...
+
   private:
     CDateTime    m_recordingTime; /*!< start time of the recording */
     bool         m_bGotMetaData;
     bool         m_bIsDeleted;    /*!< set if entry is a deleted recording which can be undelete */
     unsigned int m_iEpgEventId;   /*!< epg broadcast id associated with this recording */
     int          m_iChannelUid;   /*!< channel uid associated with this recording */
     bool         m_bRadio;        /*!< radio or tv recording */
+    int          m_iGenreType = 0;    /*!< genre type */
+    int          m_iGenreSubType = 0; /*!< genre subtype */
```
Comment (MartijnKaijser) [inline xbmc/pvr/recordings/PVRRecording.h]: No need to align them

**Review by ksooo — APPROVED** [review]
looks good to me.

**Comment by tgez** [comment]
Hello Alwin,

You can download a beta version of Kodi Chromium web browser as an addon so we can test it ?
Thank you very much for your effort

tgez

**Comment by AlwinEsch** [comment]
Thanks a lot!

**Pull request merged**